catalog/solr/catalog-solr-cache/src/main/java/org/codice/ddf/catalog/solr/cache/impl/CacheModeQueryPlugin.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.solr.cache.impl;
+
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+
+public class CacheModeQueryPlugin implements PreQueryPlugin {
+
+  @Override
+  public QueryRequest process(QueryRequest input)
+      throws PluginExecutionException, StopProcessingException {
+    if (!"cache".equals(input.getProperties().get("mode"))) {
+      return input;
+    }
+
+    HashMap<String, Serializable> props = new HashMap<>(input.getProperties());
+    props.put("cache-sources", String.join(",", input.getSourceIds()));
+
```

Comment:
do we want any debug or trace logging here?

Reply:
I was going back and forth on that just last night before pushing up the PR.  I backed off since I had changed so much already.  I can go back and see where it would not be possible to see a side effect from some other down range log statement.

Reply:
I added more logging.